File: 2100-find-good-days-to-rob-the-bank/2100-find-good-days-to-rob-the-bank.java
class Solution {
    public List<Integer> goodDaysToRobBank(int[] sec, int time) {
        int n = sec.length;
        
        int pre[] = new int[sec.length];
        int suf[] = new int[sec.length];
        
        for(int i = 1; i < n; i++)
        {
            if(sec[i] <= sec[i-1])
            {
                pre[i] = pre[i-1]+1;
            }
            else
            {
                pre[i] = 0;
            }
        }
        
        for(int i = n-2; i >= 0; i--)
        {
            if(sec[i] <= sec[i+1])
            {
                suf[i] = suf[i+1]+1;
            }
            else
            {
                suf[i] = 0;
            }
        }
        
        ArrayList<Integer> res = new ArrayList<>();
        for(int i = 0; i < n; i++)
        {
            if(time <= pre[i] && time <= suf[i])
                res.add(i);
        }
        
        return res;
    }
}

File: 2100-find-good-days-to-rob-the-bank/2100-find-good-days-to-rob-the-bank.cpp
class Solution {
public:
    vector<int> goodDaysToRobBank(vector<int>& security, int time) 
    {
        int n = security.size();
        vector<int> prefix(n,0), sufix(n,0);
        
        for(int i = 1; i < n; i++)
        {
            if(security[i] <= security[i-1])
            {
                prefix[i] = prefix[i-1]+1;
            }
        }
        
        for(int i = n-2; i >= 0; i--)
        {
            if(security[i] <= security[i+1])
            {
                sufix[i] = sufix[i+1]+1;
            }
        }
        
        vector<int> ans;
        
        for(int i = 0; i < n; i++)
        {
            if(time <= prefix[i] and time <= sufix[i])
            {
                ans.push_back(i);
            }
        }
        
        return ans;
    }
};